src/libcollections/btree/node.rs

```diff
@@ -1425,12 +1428,18 @@ pub enum TraversalItem<K, V, E> {
 /// A traversal over a node's entries and edges
 pub type Traversal<'a, K, V> = AbsTraversal<ElemsAndEdges<Zip<slice::Iter<'a, K>,
                                                               slice::Iter<'a, V>>,
-                                                              slice::Iter<'a, Node<K, V>>>>;
+                                                          slice::Iter<'a, Node<K, V>>>>;
+
+unsafe impl<'a, K: Sync, V: Sync> Sync for Traversal<'a, K, V> {}
+unsafe impl<'a, K: Sync, V: Sync> Send for Traversal<'a, K, V> {}
```

Comment:
Hm are these required? I would expect it to fall out naturally from the structure? (doesn't look like there are any interior raw pointers). (same for the `MutTraversal` below as well)